traversal/common/Predicate.java

```diff
@@ -351,7 +351,7 @@ public Value(Encoding.ValueType valueType) {
 
             public abstract boolean apply(ARG_VAL_OP operator, AttributeVertex<?> vertex, ARG_VAL_TYPE value);
 
-            private static int compareDoubles(double first, double second) {
+            public static int compareDoubles(double first, double second) {
```

Comment:
until we have query rewriting, we have a sort of re-implementation of `Predicate` evaluation in Reasoner (only to handle the case `match $x has name "john"` which may trigger a rule `then { $x has $a }` but is only valid when `$a` happens to be type of `name` with value `= "john"`.

Reply:
so we only evaluate `=` predicates in reasoner. we have it in our roadmap to revisit this requirement already.